cupy/core/carray.cuh

```diff
@@ -3,6 +3,10 @@
 #define M_PI 3.1415926535897932384626433832795
 #endif
 
+#include <cuda_fp16.h>
```

Comment:
I will do so once.
The header file `cuda_fp16.h` might be necessary even for CUDA 8 or older to support multi-GPU training with fp16.